components/script/dom/eventsource.rs

```diff
@@ -490,13 +494,23 @@ impl EventSource {
         ROUTER.add_route(action_receiver.to_opaque(), Box::new(move |message| {
             listener.notify_fetch(message.to().unwrap());
         }));
+        let cancel_receiver = ev.canceller.borrow_mut().initialize();
         global.core_resource_thread().send(
-            CoreResourceMsg::Fetch(request, FetchChannels::ResponseMsg(action_sender, None))).unwrap();
+            CoreResourceMsg::Fetch(request, FetchChannels::ResponseMsg(action_sender, Some(cancel_receiver)))).unwrap();
         // Step 13
         Ok(ev)
     }
 }
 
+// https://html.spec.whatwg.org/multipage/#garbage-collection-2
+impl Drop for EventSource {
+    fn drop(&mut self) {
+        // If an EventSource object is garbage collected while its connection is still open,
+        // the user agent must abort any instance of the fetch algorithm opened by this EventSource.
+        self.canceller.borrow_mut().cancel();
+    }
+}
```

Comment:
Question on this, is the `Drop` of a Rust `DomRoot` going to be called when the associated `JsObject` is garbage collected in the JS engine?

Reply:
Yes.